Linh - Implement Bluesky auto-poster frontend

linh2020 opened the pull request:
# Description
This PR implements the frontend interface for the Bluesky auto-poster functionality in the HighestGoodNetworkApp.
Implements: Bluesky auto-poster module (priority: high)

# Related PRs
This frontend PR is paired with the backend PR that provides Bluesky authentication, posting, and session management.
To test this frontend PR, ensure the backend PR [#1452](https://github.com/OneCommunityGlobal/HGNRest/pull/1452) is checked out and running:

# Main changes explained:
Create BlueskyManager component: A new UI section on the Announcements dashboard for managing Bluesky connection and posts.
Add login form: Users can connect using their Bluesky handle and app password (secured session saved in backend).
Add post form: Users can write a post, upload an image or GIF, and publish it to Bluesky.
Add post feed viewer: Displays user’s most recent posts including image/video previews, like/repost counts, and post metadata.
Add delete button: Each post in the feed can be deleted directly from the UI.
Image & GIF upload support: Accepts .jpg, .png, and .gif uploads (GIF autoplay and MP4 conversion coming soon).
Session detection: Automatically checks Bluesky session status and refreshes login prompt if expired.
Drag and drop support: Users can drag images or GIFs directly into the post form.
Media preview and validation: Uploaded files are previewed, checked for file type and size before submission.

# How to test:
Check into this current backend branch
Get Bluesky App Password
To connect, you need to generate an App Password:
   Visit: https://bsky.app/settings/app-passwords
Log in with your Bluesky handle (e.g., yourname.bsky.social)
Click "Create App Password"
Copy the password and use it in the frontend UI

Use Bluesky Manager
Navigate to:
http://localhost:3000/announcements → Click on Bluesky Manager tab

Log in with handle + app password
Create new posts with or without media
View your previous posts and delete if needed

⚠️ Note:
GIF support is currently limited to upload and preview only. Full autoplay and conversion to MP4 are under active development.

linh2020 committed:
Fix: added useRef import and defined boxStyle to fix rendering errors

linh2020 committed:
Fix: added useRef import and defined boxStyle to fix rendering errors

linh2020 committed:
Fix: added useRef import and defined boxStyle to fix rendering errors

linh2020 committed:
Final commit before schedule_post feature

linh2020 committed:
Final commit before schedule_post feature

linh2020 committed:
Final commit before schedule_post feature

linh2020 committed:
Merge branch 'origin/Linh_Bluesky_autoposter_frontend' into Linh_Bluesky_autoposter_frontend

linh2020 committed:
Fix import path for styles in Announcements component

linh2020 committed:
Fix linting errors: convert to CSS modules and remove styled-jsx

linh2020 committed:
Fix linting errors: convert to CSS modules and remove styled-jsx

linh2020 committed:
Remove disallowed Announcements.css to comply with CSS Module policy

linh2020 committed:
Remove disallowed Announcements.css to comply with CSS Module policy

linh2020 committed:
Remove Announcements.css import to fix test errors

linh2020 committed:
Fix all errors and finalize Announcements components

- Resolve SonarCloud quality gate failures
- Add PropTypes validation to components
- Fix cognitive complexity issues
- Improve accessibility and CSS contrast
- Remove unused imports and variables
- Fix syntax errors in BlueskyPostDetails.jsx

linh2020 committed:
Fix all errors and finalize Announcements components

- Resolve SonarCloud quality gate failures
- Add PropTypes validation to components
- Fix cognitive complexity issues
- Improve accessibility and CSS contrast
- Remove unused imports and variables
- Fix syntax errors in BlueskyPostDetails.jsx

linh2020 committed:
Fix Sonar issues in Bluesky announcements

linh2020 committed:
Fix Sonar issues in Bluesky announcements

linh2020 committed:
Fix remaining Sonar issues in Bluesky post details

linh2020 committed:
Fix remaining Sonar issues in Bluesky post details

linh2020 committed:
Fix Bluesky upload accessibility hotspot

linh2020 committed:
Fix Bluesky upload accessibility hotspot

linh2020 committed:
Fix Sonar email validation hotspot

Anusha-Gali reviewed (changes requested):
Hi Linh,

I have provided my comments in the backend PR - do check and let me know in case of re-review. https://github.com/OneCommunityGlobal/HGNRest/pull/1452#pullrequestreview-3977659349

linh2020 committed:
feat: improve Bluesky Manager UI/UX

- Styled toggle button with active/inactive pill state and blue highlight
- Redesigned connection card with branded gradient header and App Password hint link
- Added 300-char counter with color-coded progress bar (blue -> orange -> red)
- Post button disabled when over character limit
- Connected header bar shows @handle with green live-dot and Disconnect button
- Improved post feed: avatar, emoji stats, etc

linh2020 committed:
feat: improve Bluesky Manager UI/UX

- Styled toggle button with active/inactive pill state and blue highlight
- Redesigned connection card with branded gradient header and App Password hint link
- Added 300-char counter with color-coded progress bar (blue -> orange -> red)
- Post button disabled when over character limit
- Connected header bar shows @handle with green live-dot and Disconnect button
- Improved post feed: avatar, emoji stats, etc

linh2020 committed:
merge: resolve conflicts with development - adopt tab architecture, use BlueskyPostDetails for bluesky tab

linh2020 committed:
merge: resolve conflicts with development - keep BlueskyPostDetails tab, take development version for unrelated CSS files

linh2020 committed:
fix: address SonarCloud warnings in Bluesky and related CSS modules

linh2020 committed:
fix: resolve remaining Sonar reliability issues on new code

linh2020 committed:
fix: address remaining Sonar contrast and duplicate CSS declarations

linh2020 committed:
fix: resolve latest Sonar contrast and duplicate CSS warnings

linh2020 committed:
fix: resolve remaining Sonar contrast and duplicate-style warnings

Anusha-Gali reviewed (changes requested):
Hi Linh,

I have reviewed your PR locally and the posting onto Bluesky works as per requirement. I only have two concerns:
- the surrounding page is not adapted to dark mode as well as few bluesky components as well.
- the image section, once the the post is made, the text is cleared and the post is cleared but the file name in Choose File is not cleared.

**Issue**
<img width="1264" height="519" alt="Image" src="https://github.com/user-attachments/assets/d6770ef6-5239-4bda-984e-7d4c8f1a0d0e" />

**Working Screenshots**
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/893370fa-e938-478f-ace8-0c34d429fb16" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/4fbbcfb2-fae2-4b7c-a755-a2d3221fd703" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/4f6b3025-429d-4ae7-843b-fc4455474311" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/848e7761-2aa1-4808-aace-1aa0dd500c0a" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/636c1b30-e22d-4bd3-848a-bbb4c99e33c8" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/d586b22a-5c6d-4fa3-93b2-bf8780538354" />
<img width="1264" height="757" alt="Image" src="https://github.com/user-attachments/assets/a4ba2463-2582-4c34-8844-926830520de8" />
<img width="1264" height="230" alt="Image" src="https://github.com/user-attachments/assets/a13bc45e-54ab-431f-b413-8c5f4ad757eb" />
<img width="1264" height="753" alt="Image" src="https://github.com/user-attachments/assets/b5244750-b16c-4904-9159-1b4ba13572f0" />
<img width="1264" height="753" alt="Image" src="https://github.com/user-attachments/assets/b7810870-fef7-42b0-b3a9-c1c59918165c" />
<img width="1264" height="753" alt="Image" src="https://github.com/user-attachments/assets/f5e4e748-4968-4a59-ac2c-136ba7625608" />
<img width="1264" height="753" alt="Image" src="https://github.com/user-attachments/assets/49d9a930-cea2-4a0b-b346-9b126a7287cf" />
<img width="1264" height="790" alt="Image" src="https://github.com/user-attachments/assets/e1f4d8d5-fb60-4ac0-8fa3-e70f32058a00" />
<img width="1264" height="790" alt="Image" src="https://github.com/user-attachments/assets/52349869-6b60-4bc4-a9f9-e9bb8ffd47ee" />
<img width="1264" height="790" alt="Image" src="https://github.com/user-attachments/assets/fce187c6-ada3-421b-a4db-ac37efc369ad" />
<img width="1264" height="790" alt="Image" src="https://github.com/user-attachments/assets/cfc7726a-b419-4dff-818f-fdf96bd8e3ef" />

linh2020 committed:
fix: resolve rebase conflicts for Bluesky announcements and keep tab-based composer integration

linh2020 committed:
fix: resolve rebase conflicts for Bluesky announcements and keep tab-based composer integration

linh2020 committed:
chore: remove yarn.lock - use npm as primary package manager

linh2020 committed:
fix: update GitHub Actions workflow to use npm instead of yarn

- Changed cache from 'yarn' to 'npm'
- Updated install command from 'yarn install --frozen-lockfile' to 'npm ci'
- Updated test command from 'yarn run test:changed' to 'npm run test:changed'
- Updated lint command from 'yarn run lint' to 'npm run lint'
- Resolves 'yarn.lock not found' error in CI/CD pipeline
- Aligns with project's npm-based dependency management

linh2020 committed:
fix: update all GitHub Actions workflows to use npm instead of yarn

- Updated test.yml: cache: yarn → npm, yarn install → npm ci, yarn test → npm test
- Updated deploy.yml: cache: yarn → npm, yarn install --frozen-lockfile → npm ci, yarn run build → npm run build
- Resolves 'yarn.lock not found' errors in CI/CD pipelines
- Aligns all workflows with project's npm-based dependency management
- Fixes both push and pull_request test workflows

Anusha-Gali reviewed (changes requested):
Hi Linh, 

I have reviewed your PR again and the previous issues still seem to exist even after having the latest pull. Have created a video showcasing every issue as mentioned again below. Do have a look. Thank you.

- the surrounding page is not adapted to dark mode as well as few bluesky components as well.
- the image section, once the the post is made, the text is cleared and the post is cleared but the file name in Choose File is not cleared.

https://github.com/user-attachments/assets/6044f8d8-6847-4bfe-885e-bcdd0ea4ec65

linh2020 committed:
fix(announcements): fix dark mode gaps and clear upload filename state

Apply dark mode styling for Bluesky cards, inputs, drop zone, and post list

Clear file input value after post/create image reset so selected filename is removed

Fix Weekly Progress Editor container to properly respect dark mode

Add dark mode classes for email panel container, editor, and headings

linh2020 committed:
Merge origin/development into Linh_Bluesky_autoposter_frontend

linh2020 committed:
Merge origin/development into Linh_Bluesky_autoposter_frontend

linh2020 committed:
fix: use yarn in CI workflows

one-community commented:
Thank you all, merging!
